corehq/apps/userreports/tasks.py

```diff
@@ -16,56 +16,42 @@
 
 
-@task(queue='ucr_queue', ignore_result=True, acks_late=True)
-def rebuild_indicators(indicator_config_id):
-    def _is_static(config_id):
-        return config_id.startswith(StaticDataSourceConfiguration._datasource_id_prefix)
-
-    def _get_redis_key_for_config(config):
-        if _is_static(config._id):
-            rev = 'static'
-        else:
-            rev = config._rev
-        return 'ucr_queue-{}:{}'.format(config._id, rev)
-
-    is_static = _is_static(indicator_config_id)
-    if is_static:
-        config = StaticDataSourceConfiguration.by_id(indicator_config_id)
-    else:
-        config = DataSourceConfiguration.get(indicator_config_id)
+def _is_static(config_id):
+    return config_id.startswith(StaticDataSourceConfiguration._datasource_id_prefix)
 
-    adapter = IndicatorSqlAdapter(config)
 
-    couchdb = _get_db(config.referenced_doc_type)
-    client = get_redis_client().client.get_client()
-    redis_key = _get_redis_key_for_config(config)
+def _get_config_by_id(indicator_config_id):
+    if _is_static(indicator_config_id):
+        return StaticDataSourceConfiguration.by_id(indicator_config_id)
+    else:
+        return DataSourceConfiguration.get(indicator_config_id)
 
-    if len(client.smembers(redis_key)) > 0:
-        relevant_ids = client.smembers(redis_key)
+
+def _get_redis_key_for_config(config):
+    if _is_static(config._id):
+        rev = 'static'
     else:
-        if not is_static:
-            # Save the start time now in case anything goes wrong. This way we'll be
-            # able to see if the rebuild started a long time ago without finishing.
-            config.meta.build.initiated = datetime.datetime.utcnow()
-            config.save()
-            redis_key = _get_redis_key_for_config(config)
+        rev = config._rev
+    return 'ucr_queue-{}:{}'.format(config._id, rev)
 
-        adapter.rebuild_table()
-        relevant_ids = get_doc_ids_in_domain_by_type(config.domain, config.referenced_doc_type,
-                                   database=couchdb)
-        for docs in chunked(relevant_ids, 1000):
-            client.sadd(redis_key, *docs)
+
+def _build_indicators(indicator_config_id, relevant_ids):
+    config = _get_config_by_id(indicator_config_id)
+    adapter = IndicatorSqlAdapter(config)
+    couchdb = _get_db(config.referenced_doc_type)
+    redis_client = get_redis_client().client.get_client()
+    redis_key = _get_redis_key_for_config(config)
 
     for doc in iter_docs(couchdb, relevant_ids, chunksize=500):
         try:
             # save is a noop if the filter doesn't match
             adapter.save(doc)
-            client.srem(redis_key, doc.get('_id'))
+            redis_client.srem(redis_key, doc.get('_id'))
         except DataError as e:
             logging.exception('problem saving document {} to table. {}'.format(doc['_id'], e))
 
-    if not is_static:
-        client.delete(redis_key)
+    if not _is_static(indicator_config_id):
```

Comment:
Small nitpick: I think it might make more sense if this if statement wasn't in this function since it doesn't look like it's really "building" anything. It's just marking as finished